File: p0221oneactivitystate/src/main/java/com/dovhal/p0221oneactivitystate/MainActivity.java
package com.dovhal.p0221oneactivitystate;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    public String STATE = "STATES";
    private Button btnActTwo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnActTwo = findViewById(R.id.btnActTwo);
        btnActTwo.setOnClickListener(this);
        Log.d(STATE, "Main activity: onCreate()");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(STATE, "Main activity: onRestart()");

    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(STATE, "Main activity: onStart()");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(STATE, "Main activity: onResume()");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(STATE, "Main activity: onDestroy()");

    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(STATE, "Main activity: onStop()");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(STATE, "Main activity: onPause()");
    }

    @Override
    public void onClick(View v) {
        Intent intent1 = new Intent(this, ActivityTwo.class);
        startActivity(intent1);
    }
}

File: k002diceroller/src/main/java/com/dovhal/k002diceroller/MainActivity.kt
    package com.dovhal.k002diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*
import kotlin.concurrent.schedule

    @Suppress("IMPLICIT_CAST_TO_ANY")
    class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        btnRoll.setOnClickListener(){
            rollDice()
        }
        Timer("SettingUp", false).schedule(1000) {
            /* do a dice roll when the app starts */
            rollDice()
        }

    }
        /* dice rolling */
        private fun rollDice() : Unit {
            /*create a new Dice object */
            val dice = Dice(6)
            val diceRoll = dice.roll()

            val diceImage : ImageView = findViewById(R.id.dice)


            /* determine drawable resource to dice ImageView */
            val drawableResource = when(diceRoll){
                1 -> R.drawable.dice_1
                2 -> R.drawable.dice_2
                3 -> R.drawable.dice_3
                4 -> R.drawable.dice_4
                5 -> R.drawable.dice_5
                6 -> R.drawable.dice_6
                else -> {
                    Toast.makeText(this,"Something went wrong", Toast.LENGTH_SHORT)
                            .show()
                    R.drawable.dice_1
                }
            }
            /* assign resource */
            diceImage.setImageResource(drawableResource)
            /* content description for screen readers */
            diceImage.contentDescription = diceRoll.toString()
        }
}
